### PR TITLE
Add Azure active directory connector

### PR DESCRIPTION
**Pull request by huongtn**
## Description

> Provide description here

## Type of change
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## Related issues

> Fix [#1]() 

## Migration notes

> Describe migration notes if any

## Extra info

> Add any additional information

### CODE REVIEW

**Comment by huongtn** [comment]
Already test e2e

**Comment by huongtn** [comment]
Already test e2e

File: sources/azureactivedirectory-source/src/streams/groups.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  }
+  get cursorField(): string | string[] {
+    return 'createdDateTime';
+  }
+
+  async *readRecords(
+    syncMode: SyncMode,
+    cursorField?: string[],
+    streamSlice?: Dictionary<any>
+  ): AsyncGenerator<Dictionary<any, string>, any, unknown> {
```
[the remark this answers is not in the document]
Comment (Roma-Kyrnis) [reply sources/azureactivedirectory-source/src/streams/groups.ts:36]: For User stream too

File: sources/azureactivedirectory-source/acceptance-test-config.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+connector_image: azureactivedirectory-source
+tests:
+  spec:
+    - config_path: "secrets/config.json"
+      spec_path: "secrets/spec.json"
```
Comment (Roma-Kyrnis) [inline sources/azureactivedirectory-source/acceptance-test-config.yml:5]: Update spec_path to `resources/spec.json`

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]
Add README.md file to azureactivedirectory-source folder and add source to root README.md file

File: sources/azureactivedirectory-source/src/azureactivedirectory.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    return AzureActiveDirectory.azureActiveDirectory;
+  }
+
+  private static async getAccessToken(
+    config: AzureActiveDirectoryConfig
+  ): Promise<string> {
+    const version = config.auth_version ?? DEFAULT_AUTH_VERSION;
+    return new Promise(function (resolve, reject) {
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const request = require('request');
```
[the remark this answers is not in the document]
Comment (huongtn) [reply sources/azureactivedirectory-source/src/azureactivedirectory.ts:76]: I tried to use this one but not working

File: destinations/airbyte-faros-destination/src/converters/azureactivedirectory/groups.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        : undefined;
+
+    res.push({
+      model: 'org_Team',
+      record: {
+        uid,
+        name: group.displayName,
+        description: group.description,
+        lead,
+        parentTeam: {uid: 'all_teams'}, // TODO: compute parent team
```
[the remark this answers is not in the document]
Comment (huongtn) [reply destinations/airbyte-faros-destination/src/converters/azureactivedirectory/groups.ts:36]: I follow Okta source, that is added by Matthew

File: sources/azureactivedirectory-source/src/azureactivedirectory.ts [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  }
+
+  async *getUsers(): AsyncGenerator<User> {
+    const res = await this.httpClient.get<UserResponse>('users');
+    for (const item of res.data.value) {
+      const extraUserInfo = await this.httpClient.get<UserExtraInfo>(
+        `users/${item.id}?$select=Department,postalCode,createdDateTime,identities,streetAddress`
+      );
+
+      if (extraUserInfo.status === 200) {
```
[the remark this answers is not in the document]
Comment (huongtn) [reply sources/azureactivedirectory-source/src/azureactivedirectory.ts:124]: Will got issue if not check status == 200

File: sources/azureactivedirectory-source/src/azureactivedirectory.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Roma-Kyrnis) [reply sources/azureactivedirectory-source/src/azureactivedirectory.ts]: I'll try to do it and let you know about result

File: sources/azureactivedirectory-source/src/azureactivedirectory.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Roma-Kyrnis) [reply sources/azureactivedirectory-source/src/azureactivedirectory.ts:124]: Ok

File: destinations/airbyte-faros-destination/src/converters/azureactivedirectory/groups.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Roma-Kyrnis) [reply destinations/airbyte-faros-destination/src/converters/azureactivedirectory/groups.ts:36]: Okay

File: destinations/airbyte-faros-destination/src/converters/azureactivedirectory/users.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        uid,
+        title: user.displayName,
+        level: 0,
+        joinedAt,
+        terminatedAt: null,
+        department: {uid: user.department},
+        identity: {uid: user.mail, source},
+        manager,
+        reportingChain: null,
+        location: {uid: user.streetAddress, source},
```
[the remark this answers is not in the document]
Comment (Roma-Kyrnis) [reply destinations/airbyte-faros-destination/src/converters/azureactivedirectory/users.ts]: I mean write location model. res.push({ model: 'geo_Location...',})

File: sources/azureactivedirectory-source/test_files/full_configured_catalog.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  "streams": [
+    {
+      "stream": {
+        "name": "users",
+        "json_schema": {},
+        "supported_sync_modes": [
+          "full_refresh"
+        ],
+        "source_defined_cursor": true,
+        "source_defined_primary_key": [
```
Comment (Roma-Kyrnis) [inline sources/azureactivedirectory-source/test_files/full_configured_catalog.json:11]: Update primary keys

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]
Fix destination test

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]
There are some issues. Can we fix them? 
First, when run acceptance-test, it throws: `AssertionError: Some of defined cursor fields ['renewedDateTime'] are not specified in discover schema properties for groups stream   E           assert {}`

Second, while writing geo_Location to Faros in users converter

File: destinations/airbyte-faros-destination/src/converters/azureactivedirectory/users.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    const location = user.streetAddress
+      ? {uid: user.streetAddress, source}
+      : undefined;
+
+    if (user.streetAddress) {
+      const geo_Location = {
+        uid: user.streetAddress,
+        name: user.streetAddress,
+        raw: user.streetAddress,
+        address: user.streetAddress,
```
Comment (Roma-Kyrnis) [inline destinations/airbyte-faros-destination/src/converters/azureactivedirectory/users.ts:60]: Write geo_Address model too

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]

File: destinations/airbyte-faros-destination/src/converters/azureactivedirectory/users.ts (hunk 8 above)
Comment (Roma-Kyrnis) [reply destinations/airbyte-faros-destination/src/converters/azureactivedirectory/users.ts:60]: https://github.com/faros-ai/canonical-models/blob/main/resources/models/geo.gql

**Comment by huongtn** [comment]
> There are some issues. Can we fix them? First, when run acceptance-test, it throws: `AssertionError: Some of defined cursor fields ['renewedDateTime'] are not specified in discover schema properties for groups stream E assert {}`
> 
> Second, while writing geo_Location to Faros in users converter

ok I fixed

**Review by Roma-Kyrnis — APPROVED** [review]

File: sources/azureactivedirectory-source/src/azureactivedirectory.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Roma-Kyrnis) [reply sources/azureactivedirectory-source/src/azureactivedirectory.ts]: We can use Axios request with querystring params. `[Example](https://stackoverflow.com/questions/67502700/axios-returns-404-for-a-post-request-with-form-data)`

**Review by Roma-Kyrnis — CHANGES_REQUESTED** [review]

File: sources/azureactivedirectory-source/src/azureactivedirectory.ts [hunk 9]
```diff
@@ -70,24 +70,18 @@ export class AzureActiveDirectory {
...
-
-      axios
-        .post(`${AUTH_URL}/${config.tenant_id}/oauth2/${version}/token`, data)
-        .then((response) => {
-          resolve(response.data.access_token);
-        })
-        .catch((err) => {
-          reject(err);
-        });
+    const data = new URLSearchParams({
```
Comment (Roma-Kyrnis) [inline sources/azureactivedirectory-source/src/azureactivedirectory.ts:73]: Great job!

**Comment by tovbinm** [comment]
@cjwooo we might need to bump the version

**Pull request merged**